var/spack/repos/builtin/packages/py-numpy/package.py

```diff
@@ -90,28 +91,37 @@ class PyNumpy(PythonPackage):
     variant("blas", default=True, description="Build with BLAS support")
     variant("lapack", default=True, description="Build with LAPACK support")
 
-    # Based on wheel availability on PyPI
-    depends_on("python@3.9:3.11", when="@1.25:", type=("build", "link", "run"))
+    depends_on("python@3.9:3.12", when="@2.26:", type=("build", "link", "run"))
+    depends_on("python@3.9:3.11", when="@1.25", type=("build", "link", "run"))
     depends_on("python@3.8:3.11", when="@1.23.2:1.24", type=("build", "link", "run"))
     depends_on("python@3.8:3.10", when="@1.22:1.23.1", type=("build", "link", "run"))
     depends_on("python@:3.10", when="@1.21.2:1.21", type=("build", "link", "run"))
     depends_on("python@:3.9", when="@1.19.3:1.21.1", type=("build", "link", "run"))
     depends_on("python@:3.8", when="@1.17.3:1.19.2", type=("build", "link", "run"))
     depends_on("python@:3.7", when="@1.14.5:1.17.2", type=("build", "link", "run"))
 
-    # https://github.com/spack/spack/pull/32078
-    depends_on("py-setuptools@:63", type=("build", "run"))
-    depends_on("py-setuptools@:59", when="@:1.22.1", type=("build", "run"))
-    # Check pyproject.toml for updates to the required cython version
+    depends_on("py-cython@0.29.34:3.0", when="@1.26:", type="build")
     depends_on("py-cython@0.29.34:2", when="@1.25:", type="build")
     depends_on("py-cython@0.29.13:2", when="@1.18.0:", type="build")
     depends_on("py-cython@0.29.14:2", when="@1.18.1:", type="build")
     depends_on("py-cython@0.29.21:2", when="@1.19.1:", type="build")
     depends_on("py-cython@0.29.24:2", when="@1.21.2:", type="build")
     depends_on("py-cython@0.29.30:2", when="@1.22.4:", type="build")
+    depends_on("py-pyproject-metadata@0.7.1:", when="@1.26:", type="build")
+    depends_on("py-tomli@1:", when="@1.26: ^python@:3.10", type="build")
+    depends_on("py-setuptools@60:", when="@1.26: ^python@3.12:", type="build")
+    # https://github.com/spack/spack/pull/32078
+    depends_on("py-setuptools@:63", when="@:1.25", type=("build", "run"))
+    depends_on("py-setuptools@:59", when="@:1.22.1", type=("build", "run"))
+    depends_on("py-colorama", when="@1.26: platform=windows", type="build")
+
+    # meson is vendored, ninja and pkgconfig are not
+    depends_on("ninja@1.8.2:", when="@1.26:", type="build")
+    depends_on("pkgconfig", when="@1.26:", type="build")
     depends_on("blas", when="+blas")
     depends_on("lapack", when="+lapack")
 
+    # test_requirements.txt
```

Comment:
May need to update these. I had several tests fail:
```
FAILED ../linalg/tests/test_linalg.py::TestDet::test_types[complex64] - Runti...
FAILED ../linalg/tests/test_linalg.py::TestDet::test_types[complex128] - Runt...
FAILED ../random/tests/test_extending.py::test_cython - FileNotFoundError: [E...
FAILED ../tests/test_public_api.py::test_api_importable - UserWarning: Distut...
ERROR ../core/tests/test_mem_policy.py::test_set_policy - RuntimeError: could...
ERROR ../core/tests/test_mem_policy.py::test_default_policy_singleton - Runti...
ERROR ../core/tests/test_mem_policy.py::test_policy_propagation - RuntimeErro...
ERROR ../core/tests/test_mem_policy.py::test_context_locality - RuntimeError:...
ERROR ../core/tests/test_mem_policy.py::test_thread_locality - RuntimeError: ...
ERROR ../core/tests/test_mem_policy.py::test_switch_owner[0] - RuntimeError: ...
ERROR ../core/tests/test_mem_policy.py::test_switch_owner[1] - RuntimeError: ...
ERROR ../core/tests/test_mem_policy.py::test_switch_owner[None] - RuntimeErro...
ERROR ../core/tests/test_mem_policy.py::test_owner_is_base - RuntimeError: co...
```

Reply:
The `test_extending.py::test_cython` one is because the test is now run with `meson`/`ninja`. 

The `TestDet` ones could be failures due to OpenBLAS perhaps. Are those with OpenBLAS 0.3.23/24?

`test_public_api.py`: not sure, I haven't seen that fail before.

`test_mem_policy.py` has been fragile, but I can't tell from this comment what is going wrong. It does require rebuilding, so perhaps also `meson`/`ninja`?

Reply:
> Are those with OpenBLAS 0.3.23/24?

0.3.21